Project Composition (#1584)
bonitasoft/bonita-doc: modules/ROOT/pages/software-extensibility.adoc

=== Connectors

In Community, Teamwork, Efficiency, Performance and Enterprise editions

Bonita comes with more than 80 xref:connectivity-overview.adoc[standard connectors] to the major information system components: major databases (Oracle, Microsoft, Postgres, etc.), SOAP Webservice, Salesforce, Email, etc.

> A new connector enables you to provide new connectivity capabilities for processes.

Reviewer: We should also talk about the connectors that do not connect anywhere but are tools that need to be created as connectors: Word templating, Word to pdf transformation, script...

> A new connector enables you to provide new connectivity capabilities for processes.

Reviewer: PostgreSQL

> 

Reviewer: To use one of them in the processes, go to the Extensions view and select the connector in the "Marketplace".
Then, configure it at task level in the "Execution" tab.

> Connectors are developped outside the Studio with a dedicated maven archetype. Consult this page for more information: xref:connector-archetype.adoc[Create a connector].

Reviewer: Maven

> Custom connectors are added to a Bonita project with the xref:managing_extension_studio.adoc[extension view].

Reviewer 2: ```suggestion
Custom connectors are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].
```

> Custom connectors are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].

Reviewer: Extensions view

=== Actor filters
However you create your own connectors to have new connectivity capabilities for processes.

In Community, Teamwork, Efficiency, Performance and Enterprise editions
Connectors are developped outside the Studio with a dedicated maven archetype. Consult this page for more information: xref:connector-archetype.adoc[Create a connector].

Custom connectors are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].

=== Actor filters

Bonita comes with a set of xref:actor-filtering.adoc[standard actor filters] that can be used to reduce the list of candidates for process tasks.

> A new actor filter enables you to provide new filtering capabilities to processes. To xref:creating-an-actor-filter.adoc[implement a new actor filter],

Reviewer: To use one of them in the processes, go to the Extensions view and select the actor filter in the "Marketplace".
Then, configure it at human task level in the "General" tab.

> Actor filters are developped outside the Studio with a dedicated maven archetype. Consult this page for more information:xref:actor-filter-archetype.adoc[Create an actor filter]

Reviewer: However you create your own actor filters to select new organization entities to be candidates for human tasks.

> Custom actor filters are added to a Bonita project the xref:managing_extension_studio.adoc[extension view].

Reviewer 2: ```suggestion
Custom actor filters are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].
```

=== REST API

The xref:rest-api-overview.adoc[REST APIs] enable you to integrate your Bonita processes into your application and execute operations on Bonita objects including business objects.

> Pages and forms created with the UI designer rely on the REST APIs to manage data.

Reviewer: UI Designer

> In addition to the standard APIs and resources, you can define xref:rest-api-extensions.adoc[REST API extensions].

Reviewer: What are the resources referred at?

> REST API Extension can either created in the xref:rest-api-extensions.adoc[Studio] or with the xref:rest-api-extension-archetype.adoc[dedicated maven archetype].

Reviewer 2: ```suggestion
REST API Extension can either be created in the xref:rest-api-extensions.adoc[Studio] or with the xref:rest-api-extension-archetype.adoc[dedicated maven archetype].
```

> REST API Extension can either be created in the xref:rest-api-extensions.adoc[Studio] or with the xref:rest-api-extension-archetype.adoc[dedicated maven archetype].

Reviewer: A REST API extension

> REST API Extension can either be created in the xref:rest-api-extensions.adoc[Studio] or with the xref:rest-api-extension-archetype.adoc[dedicated maven archetype].

Reviewer: dedicated Maven archetype

> Custom REST API extension developed with the maven archetype are added to a Bonita project with the xref:managing_extension_studio.adoc[extension view].

Reviewer 2: ```suggestion
Custom REST API extensions developed with the maven archetype are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].
```

> Custom REST API extensions developed with the maven archetype are added to a Bonita project from the xref:managing_extension_studio.adoc[extension view].

Reviewer: REST API extensions developed with the Maven archetype are added to a Bonita project from the xref:managing_extension_studio.adoc[Extensions view].

=== Custom widgets

> If the standard UI designer xref:widgets.adoc[widgets] do not meet your needs, you can create a xref:custom-widgets.adoc[custom widget].

Reviewer: UI Designer

> You can then use your custom widget in pages, forms, and (for Subscription editions) fragments.

Reviewer: Fragments are also for Community now

=== Living Application Themes

> Custom theme are developped outside the Studio with a dedicated maven archetype. Consult this page for more information:xref:customize-living-application-theme.adoc[Create an theme]

Reviewer 2: ```suggestion
Custom themes are developed outside the Studio with a dedicated maven archetype. Consult this page for more information: xref:customize-living-application-theme.adoc[Create an theme]
```

> Custom themes are developed outside the Studio with a dedicated maven archetype. Consult this page for more information: xref:customize-living-application-theme.adoc[Create an theme]

Reviewer: Maven
